packages/web-client/tests/acceptance/network-problems-test.ts

```diff
@@ -0,0 +1,101 @@
+import { module, test } from 'qunit';
+import { settled, visit } from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+import Layer2TestWeb3Strategy from '@cardstack/web-client/utils/web3-strategies/test-layer2';
+import Layer1TestWeb3Strategy from '@cardstack/web-client/utils/web3-strategies/test-layer1';
+
+import { createDepotSafe } from '@cardstack/web-client/utils/test-factories';
+
+const MODAL = '[data-test-network-problem-modal]';
+const MODAL_TITLE = '[data-test-network-problem-modal-title]';
+const MODAL_BODY = '[data-test-network-problem-modal-body]';
+const MODAL_ACTION = '[data-test-network-problem-modal-action]';
+
+const MODAL_IS_DISMISSABLE_ATTRIBUTE =
+  'data-test-network-problem-modal-dismissable';
+
+module('Acceptance | network-problems', function (hooks) {
+  setupApplicationTest(hooks);
+  let layer2AccountAddress = '0x182619c6Ea074C053eF3f1e1eF81Ec8De6Eb6E44';
+  let layer2Service: Layer2TestWeb3Strategy;
+  let layer1Service: Layer1TestWeb3Strategy;
+
+  hooks.beforeEach(async function () {
+    layer2Service = this.owner.lookup('service:layer2-network')
+      .strategy as Layer2TestWeb3Strategy;
+    layer2Service.test__simulateRemoteAccountSafes(layer2AccountAddress, [
+      createDepotSafe({}),
+    ]);
+    await layer2Service.test__simulateAccountsChanged([layer2AccountAddress]);
+
+    let layer1AccountAddress = '0xaCD5f5534B756b856ae3B2CAcF54B3321dd6654Fb6';
+    layer1Service = this.owner.lookup('service:layer1-network')
+      .strategy as Layer1TestWeb3Strategy;
+    layer1Service.test__simulateAccountsChanged(
+      [layer1AccountAddress],
+      'metamask'
+    );
+
+    await visit('/card-pay/balances');
+  });
+
+  test('Layer 2 incorrect chain event triggers the showing of a modal', async function (assert) {
+    layer2Service.simpleEmitter.emit('incorrect-chain');
+
+    await settled();
+
+    assert.dom(MODAL_TITLE).containsText('Please connect to L2 test chain');
```

Comment:
I think most tests use the `currentNetworkDisplayInfo` abstraction to generate the strings for comparison… I actually kind of like having the bare string, just wanted to highlight the possible discrepancy though 🤓